<h5 font-family:Arial;">Caitlyn</h5>
<h5 font-family:Arial;">10/07/2025</h5>
<h5 font-family:Arial;">Project #1, Part 2: Data Exploration and Cleaning</h5>
<h5 font-family:Arial;">Non-Majors Survey Results dataset cleaning and tidying.</h5>

<h5 font-weight:bold; font-family:Arial;">Imports</h5>


In [1]:
import pandas as pd

<h5 font-weight:bold; font-family:Arial;">Data Loading</h5>


In [2]:
df = pd.read_csv("../data/original_data2024.csv")

<h5 font-weight:bold; font-family:Arial;">Exploring the dataset</h5>


<h5 font-family:Arial;">Before Cleaning (After cleaning at the end)</h5>

In [3]:
df.head()

Timestamp   Which course are you currently enrolled in?  \
0  2024/09/10 10:47:35 AM AST  CMP 126 Computer Technology and Applications   
1  2024/09/10 11:00:05 AM AST  CMP 126 Computer Technology and Applications   
2  2024/09/10 11:02:51 AM AST  CMP 126 Computer Technology and Applications   
3  2024/09/10 11:05:25 AM AST  CMP 126 Computer Technology and Applications   
4   2024/09/10 8:37:36 PM AST         CMP 101 Computer Information Literacy   

  What motivated you to seek a computing class at CCM? [It’s a required class for the degree I’m seeking]  \
0                                                 No                                                        
1                                                Yes                                                        
2                                                Yes                                                        
3                                                 No                                                        
4                                                Yes                                                        

  What motivated you to seek a computing class at CCM? [To keep current in computing skills]  \
0                                                Yes                                           
1                                                Yes                                           
2                                                Yes                                           
3                                                Yes                                           
4                                                Yes                                           

  What motivated you to seek a computing class at CCM? [Career Advancement]  \
0                                                Yes                          
1                                                Yes                          
2                                                Yes                          
3                                                 No                          
4                                                Yes                          

  What motivated you to seek a computing class at CCM? [Career Change]  \
0                                                Yes                     
1                                                 No                     
2                                                Yes                     
3                                                 No                     
4                                                 No                     

  What motivated you to seek a computing class at CCM? [Professional Development]  \
0                                                Yes                                
1                                                Yes                                
2                                                Yes                                
3                                                 No                                
4                                                Yes                                

  What motivated you to seek a computing class at CCM? [Job Displacement]  \
0                                                 No                        
1                                                Yes                        
2                                                Yes                        
3                                                 No                        
4                                                 No                        

  What motivated you to seek a computing class at CCM? [Relocation]  \
0                                                 No                  
1                                                 No                  
2                                                Yes                  
3                                                 No                  
4                                                 No            

In [4]:
df.tail()

Timestamp   Which course are you currently enrolled in?  \
87  2024/09/20 10:55:34 PM AST         CMP 101 Computer Information Literacy   
88  2024/09/20 11:02:45 PM AST  CMP 126 Computer Technology and Applications   
89  2024/09/20 11:07:16 PM AST  CMP 126 Computer Technology and Applications   
90   2024/09/21 5:06:23 PM AST         CMP 101 Computer Information Literacy   
91  2024/09/22 10:08:19 PM AST  CMP 126 Computer Technology and Applications   

   What motivated you to seek a computing class at CCM? [It’s a required class for the degree I’m seeking]  \
87                                                Yes                                                        
88                                                Yes                                                        
89                                                Yes                                                        
90                                                 No                                                        
91                                                Yes                                                        

   What motivated you to seek a computing class at CCM? [To keep current in computing skills]  \
87                                                Yes                                           
88                                                Yes                                           
89                                                 No                                           
90                                                Yes                                           
91                                                Yes                                           

   What motivated you to seek a computing class at CCM? [Career Advancement]  \
87                                                Yes                          
88                                                Yes                          
89                                                 No                          
90                                                 No                          
91                                                Yes                          

   What motivated you to seek a computing class at CCM? [Career Change]  \
87                                                Yes                     
88                                                 No                     
89                                                 No                     
90                                                 No                     
91                                                 No                     

   What motivated you to seek a computing class at CCM? [Professional Development]  \
87                                                 No                                
88                                                Yes                                
89                                                Yes                                
90                                                 No                                
91                                                Yes                                

   What motivated you to seek a computing class at CCM? [Job Displacement]  \
87                                                 No                        
88                                                Yes                        
89                                                 No                        
90                                                 No                        
91                                                 No                        

   What motivated you to seek a computing class at CCM? [Relocation]  \
87                                                 No                  
88                                                 No                  
89                                                 No                  
90                                                 No                  
91                

In [5]:
df.describe()

On a scale of 1 to 5, with 1 being not at all interested and 5 being extremely interested, how interested are you taking more courses in Computer Science, Information Technology or Game Development?
count                                          92.000000                                                                                                                                                     
mean                                            2.576087                                                                                                                                                     
std                                             1.131310                                                                                                                                                     
min                                             1.000000                                                                                                                                                     
25%                                             2.000000                                                                                                                                                     
50%                                             3.000000                                                                                                                                                     
75%                                             3.000000                                                                                                                                                     
max                                             5.000000

<h5 font-family:Arial;">Based on the data, students show relatively low interest in taking additional CS/IT/Game Development courses, with an average rating of 2.58 out of 5 and three-quarters of respondents rating their interest at 3 (neutral) or below.</h5>


In [6]:
df.columns

Index(['Timestamp', 'Which course are you currently enrolled in?',
       'What motivated you to seek a computing class at CCM? [It’s a required class for the degree I’m seeking]',
       'What motivated you to seek a computing class at CCM? [To keep current in computing skills]',
       'What motivated you to seek a computing class at CCM? [Career Advancement]',
       'What motivated you to seek a computing class at CCM? [Career Change]',
       'What motivated you to seek a computing class at CCM? [Professional Development]',
       'What motivated you to seek a computing class at CCM? [Job Displacement]',
       'What motivated you to seek a computing class at CCM? [Relocation]',
       'What motivated you to seek a computing class at CCM? [IT Industry Certifications]',
       'What motivated you to seek a computing class at CCM? [Financial]',
       'What motivated you to seek a computing class at CCM? [Personal Enrichment]',
       'What motivated you to seek a computing class at

In [7]:
df.shape

(92, 71)

<h5 font-family:Arial;">92 rows,71 columns</h5>

In [8]:
df.dtypes

Timestamp                                                                                                                                    object
Which course are you currently enrolled in?                                                                                                  object
What motivated you to seek a computing class at CCM? [It’s a required class for the degree I’m seeking]                                      object
What motivated you to seek a computing class at CCM? [To keep current in computing skills]                                                   object
What motivated you to seek a computing class at CCM? [Career Advancement]                                                                    object
                                                                                                                                              ...  
If you answered that you were interested in taking more computing classes, which ones interest you most? [Game D

In [9]:
df.sample

<bound method NDFrame.sample of                      Timestamp   Which course are you currently enrolled in?  \
0   2024/09/10 10:47:35 AM AST  CMP 126 Computer Technology and Applications   
1   2024/09/10 11:00:05 AM AST  CMP 126 Computer Technology and Applications   
2   2024/09/10 11:02:51 AM AST  CMP 126 Computer Technology and Applications   
3   2024/09/10 11:05:25 AM AST  CMP 126 Computer Technology and Applications   
4    2024/09/10 8:37:36 PM AST         CMP 101 Computer Information Literacy   
..                         ...                                           ...   
87  2024/09/20 10:55:34 PM AST         CMP 101 Computer Information Literacy   
88  2024/09/20 11:02:45 PM AST  CMP 126 Computer Technology and Applications   
89  2024/09/20 11:07:16 PM AST  CMP 126 Computer Technology and Applications   
90   2024/09/21 5:06:23 PM AST         CMP 101 Computer Information Literacy   
91  2024/09/22 10:08:19 PM AST  CMP 126 Computer Technology and Applications   

   What

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 92 entries, 0 to 91
Data columns (total 71 columns):
 #   Column                                                                                                                                                                                                                                           Non-Null Count  Dtype 
---  ------                                                                                                                                                                                                                                           --------------  ----- 
 0   Timestamp                                                                                                                                                                                                                                        92 non-null     object
 1   Which course are you currently enrolled in?                                          

<h5 font-family:Arial;">
Many questions were split into multiple columns due to checkbox responses (e.g., each motivation option has its own column).  
Several columns contain missing values, and most column names are long, with extra spaces, punctuation, or bracketed text.  
Data types are mostly objects, with one numeric column for the interest scale.  
Overall, the dataset needs cleaning for clearer analysis.
</h5>


<h5 font-weight:bold; font-family:Arial;">Data Cleaning</h5>

<h5 font-family:Arial;">Renaming Columns - improves readability and organization</h5>

In [11]:
df.columns

Index(['Timestamp', 'Which course are you currently enrolled in?',
       'What motivated you to seek a computing class at CCM? [It’s a required class for the degree I’m seeking]',
       'What motivated you to seek a computing class at CCM? [To keep current in computing skills]',
       'What motivated you to seek a computing class at CCM? [Career Advancement]',
       'What motivated you to seek a computing class at CCM? [Career Change]',
       'What motivated you to seek a computing class at CCM? [Professional Development]',
       'What motivated you to seek a computing class at CCM? [Job Displacement]',
       'What motivated you to seek a computing class at CCM? [Relocation]',
       'What motivated you to seek a computing class at CCM? [IT Industry Certifications]',
       'What motivated you to seek a computing class at CCM? [Financial]',
       'What motivated you to seek a computing class at CCM? [Personal Enrichment]',
       'What motivated you to seek a computing class at

In [12]:
df.columns = [
    'timestamp',
    'current_course',
    'motivation_required_class',
    'motivation_keep_current',
    'motivation_career_advancement',
    'motivation_career_change',
    'motivation_professional_development',
    'motivation_job_displacement',
    'motivation_relocation',
    'motivation_certifications',
    'motivation_financial',
    'motivation_enrichment',
    'motivation_curiosity',
    'precollege_open_house',
    'precollege_instant_decision',
    'precollege_info_session',
    'precollege_titans_tuesday',
    'precollege_women_who_dare',
    'precollege_college_fair',
    'impact_school_class',
    'impact_school_club',
    'impact_competitions',
    'impact_afterschool_program',
    'impact_summer_camp',
    'impact_ap_class',
    'impact_dual_enrollment',
    'impact_family_friend_influence',
    'impact_family_friend_in_field',
    'impact_teacher_counselor',
    'impact_employer',
    'heard_website',
    'heard_billboard',
    'heard_tv',
    'heard_radio',
    'heard_social_media',
    'heard_event',
    'heard_family_friend',
    'heard_employer',
    'heard_current_student',
    'heard_alumni',
    'heard_high_school',
    'attend_affordable_cost',
    'attend_location_convenience',
    'attend_program_choice',
    'attend_online_offerings',
    'attend_family_friend_referral',
    'attend_faculty_staff',
    'attend_reputation',
    'attend_scholarships',
    'attend_transfer_student',
    'attend_small_classes',
    'attend_extracurricular',
    'degree_program',
    'info_high_school_counselor',
    'info_it_website',
    'info_admissions',
    'info_ccm_advisor',
    'info_employer',
    'info_other',
    'interest_level',
    'interest_web_dev',
    'interest_cybersecurity',
    'interest_app_dev',
    'interest_data_analytics',
    'interest_machine_learning',
    'interest_programming',
    'interest_game_design',
    'interest_hardware_repair',
    'gender',
    'race_ethnicity',
    'age'
]


In [13]:
df.columns

Index(['timestamp', 'current_course', 'motivation_required_class',
       'motivation_keep_current', 'motivation_career_advancement',
       'motivation_career_change', 'motivation_professional_development',
       'motivation_job_displacement', 'motivation_relocation',
       'motivation_certifications', 'motivation_financial',
       'motivation_enrichment', 'motivation_curiosity',
       'precollege_open_house', 'precollege_instant_decision',
       'precollege_info_session', 'precollege_titans_tuesday',
       'precollege_women_who_dare', 'precollege_college_fair',
       'impact_school_class', 'impact_school_club', 'impact_competitions',
       'impact_afterschool_program', 'impact_summer_camp', 'impact_ap_class',
       'impact_dual_enrollment', 'impact_family_friend_influence',
       'impact_family_friend_in_field', 'impact_teacher_counselor',
       'impact_employer', 'heard_website', 'heard_billboard', 'heard_tv',
       'heard_radio', 'heard_social_media', 'heard_event',
   

<h5 font-weight:bold; font-family:Arial;">Removing Irrelavant Columns</h5>


<h5 font-family:Arial;">Survey questions that answer the 5 data analysis questions: 1, 2, 3, 5, 7, 9, 10, 11, 12, 13</h5>

In [14]:
list(enumerate(df.columns))


[(0, 'timestamp'),
 (1, 'current_course'),
 (2, 'motivation_required_class'),
 (3, 'motivation_keep_current'),
 (4, 'motivation_career_advancement'),
 (5, 'motivation_career_change'),
 (6, 'motivation_professional_development'),
 (7, 'motivation_job_displacement'),
 (8, 'motivation_relocation'),
 (9, 'motivation_certifications'),
 (10, 'motivation_financial'),
 (11, 'motivation_enrichment'),
 (12, 'motivation_curiosity'),
 (13, 'precollege_open_house'),
 (14, 'precollege_instant_decision'),
 (15, 'precollege_info_session'),
 (16, 'precollege_titans_tuesday'),
 (17, 'precollege_women_who_dare'),
 (18, 'precollege_college_fair'),
 (19, 'impact_school_class'),
 (20, 'impact_school_club'),
 (21, 'impact_competitions'),
 (22, 'impact_afterschool_program'),
 (23, 'impact_summer_camp'),
 (24, 'impact_ap_class'),
 (25, 'impact_dual_enrollment'),
 (26, 'impact_family_friend_influence'),
 (27, 'impact_family_friend_in_field'),
 (28, 'impact_teacher_counselor'),
 (29, 'impact_employer'),
 (30, 'h

In [15]:
df = df.drop(df.columns[[ 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29,
    41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51,
    53, 54, 55, 56, 57, 58]], axis=1)

In [16]:
list(enumerate(df.columns))

[(0, 'timestamp'),
 (1, 'current_course'),
 (2, 'motivation_required_class'),
 (3, 'motivation_keep_current'),
 (4, 'motivation_career_advancement'),
 (5, 'motivation_career_change'),
 (6, 'motivation_professional_development'),
 (7, 'motivation_job_displacement'),
 (8, 'motivation_relocation'),
 (9, 'motivation_certifications'),
 (10, 'motivation_financial'),
 (11, 'motivation_enrichment'),
 (12, 'motivation_curiosity'),
 (13, 'precollege_open_house'),
 (14, 'precollege_instant_decision'),
 (15, 'precollege_info_session'),
 (16, 'precollege_titans_tuesday'),
 (17, 'precollege_women_who_dare'),
 (18, 'precollege_college_fair'),
 (19, 'heard_website'),
 (20, 'heard_billboard'),
 (21, 'heard_tv'),
 (22, 'heard_radio'),
 (23, 'heard_social_media'),
 (24, 'heard_event'),
 (25, 'heard_family_friend'),
 (26, 'heard_employer'),
 (27, 'heard_current_student'),
 (28, 'heard_alumni'),
 (29, 'heard_high_school'),
 (30, 'degree_program'),
 (31, 'interest_level'),
 (32, 'interest_web_dev'),
 (33, '

<h5 font-weight:bold; font-family:Arial;">Cleaning Categorical Values</h5>


<h5 font-family:Arial;">Organizing the degree programs and creating new categories where necessary.</h5>

In [17]:
df['degree_program'].unique()

array(['Liberal Arts', 'child psychology ',
       'Broadcasting Arts and Technology', 'analyst', 'Journalism',
       'Early Childhood Education', 'Nursing', 'Radiography',
       'Exercise Science', 'Engineering Science', 'Music Technology',
       'Nursing, but not at CCM - I am a visiting student ', 'Psychology',
       'Criminal Justice', 'Not in a degree program', 'Biology',
       'Culinary Arts and Sciences', 'History Education',
       'Biology - Environmental Track', 'Communication',
       'Child and Family Studies', 'Musical Theatre', 'Fire Science '],
      dtype=object)

In [18]:
major_mapping = {
    # --- Direct matches from the form ---
    'Broadcasting Arts and Technology': 'Broadcasting Arts and Technology',
    'Child and Family Studies': 'Child and Family Studies',
    'Communication': 'Communication',
    'Criminal Justice': 'Criminal Justice',
    'Early Childhood Education': 'Early Childhood Education',
    'Exercise Science': 'Exercise Science',
    'Journalism': 'Journalism',
    'Liberal Arts': 'Liberal Arts',
    'Music Technology': 'Music Technology',
    'Nursing': 'Nursing',
    'Radiography': 'Radiography',
    'Not in a degree program': 'Not in a degree program',

    # --- Slight variations or clarifications ---
    'Nursing, but not at CCM - I am a visiting student ': 'Nursing',
    'Musical Theatre': 'Music Technology',  # fits under same creative arts area

    # --- Related / similar but not exact form options ---
    'Engineering Science': 'Engineering',
    'Biology': 'Math / Science',
    'Biology - Environmental Track': 'Math / Science',
    'Culinary Arts and Sciences': 'Hospitality / Culinary',
    'History Education': 'Liberal Arts',
    'Psychology': 'Psychology / Social Science',
    'child psychology ': 'Psychology / Social Science',
    'analyst': 'Computer / Technology',  # could indicate Data / IT-related field

    # --- Default fallback ---
    'Other': 'Other'
}


In [19]:
df['major_mapped'] = df['degree_program'].map(major_mapping)

In [20]:
print(df['major_mapped'].unique())

['Liberal Arts' 'Psychology / Social Science'
 'Broadcasting Arts and Technology' 'Computer / Technology' 'Journalism'
 'Early Childhood Education' 'Nursing' 'Radiography' 'Exercise Science'
 'Engineering' 'Music Technology' 'Criminal Justice'
 'Not in a degree program' 'Math / Science' 'Hospitality / Culinary'
 'Communication' 'Child and Family Studies' nan]


<h5 font-weight:bold; font-family:Arial;">Checking for duplicate entries</h5>

In [21]:
print(f"Number of duplicates: {df.duplicated().sum()}")

Number of duplicates: 0


<h5 font-weight:bold; font-family:Arial;">After Cleaning</h5>

In [22]:
df.head()

timestamp                                current_course  \
0  2024/09/10 10:47:35 AM AST  CMP 126 Computer Technology and Applications   
1  2024/09/10 11:00:05 AM AST  CMP 126 Computer Technology and Applications   
2  2024/09/10 11:02:51 AM AST  CMP 126 Computer Technology and Applications   
3  2024/09/10 11:05:25 AM AST  CMP 126 Computer Technology and Applications   
4   2024/09/10 8:37:36 PM AST         CMP 101 Computer Information Literacy   

  motivation_required_class motivation_keep_current  \
0                        No                     Yes   
1                       Yes                     Yes   
2                       Yes                     Yes   
3                        No                     Yes   
4                       Yes                     Yes   

  motivation_career_advancement motivation_career_change  \
0                           Yes                      Yes   
1                           Yes                       No   
2                           Yes                      Yes   
3                            No                       No   
4                           Yes                       No   

  motivation_professional_development motivation_job_displacement  \
0                                 Yes                          No   
1                                 Yes                         Yes   
2                                 Yes                         Yes   
3                                  No                          No   
4                                 Yes                          No   

  motivation_relocation motivation_certifications  ... interest_app_dev  \
0                    No                        No  ...               No   
1                    No                        No  ...              Yes   
2                   Yes                        No  ...              Yes   
3                    No                        No  ...               No   
4                    No                        No  ...               No   

  interest_data_analytics interest_machine_learning interest_programming  \
0                      No                        No                   No   
1                     Yes                       Yes                  Yes   
2                      No                       Yes                  Yes   
3                     Yes                        No                   No   
4                      No                        No                   No   

  interest_game_design interest_hardware_repair             gender  \
0                   No                       No  Prefer not to say   
1                   No                       No              Woman   
2                  Yes                      Yes                Man   
3                  Yes                       No                Man   
4                   No                       No              Woman   

                                      race_ethnicity             age  \
0  Hispanic or Latino;Black/African American;Whit...           19-20   
1                                 Hispanic or Latino           19-20   
2                                Choose not to reply  18 and younger   
3                                 Hispanic or Latino           19-20   
4                                    White/Caucasian           19-20   

                       major_mapped  
0                      Liberal Arts  
1       Psychology / Social Science  
2  Broadcasting Arts and Technology  
3             Computer / Technology  
4                        Journalism  

[5 rows x 44 columns]

In [23]:
df.tail()

timestamp                                current_course  \
87  2024/09/20 10:55:34 PM AST         CMP 101 Computer Information Literacy   
88  2024/09/20 11:02:45 PM AST  CMP 126 Computer Technology and Applications   
89  2024/09/20 11:07:16 PM AST  CMP 126 Computer Technology and Applications   
90   2024/09/21 5:06:23 PM AST         CMP 101 Computer Information Literacy   
91  2024/09/22 10:08:19 PM AST  CMP 126 Computer Technology and Applications   

   motivation_required_class motivation_keep_current  \
87                       Yes                     Yes   
88                       Yes                     Yes   
89                       Yes                      No   
90                        No                     Yes   
91                       Yes                     Yes   

   motivation_career_advancement motivation_career_change  \
87                           Yes                      Yes   
88                           Yes                       No   
89                            No                       No   
90                            No                       No   
91                           Yes                       No   

   motivation_professional_development motivation_job_displacement  \
87                                  No                          No   
88                                 Yes                         Yes   
89                                 Yes                          No   
90                                  No                          No   
91                                 Yes                          No   

   motivation_relocation motivation_certifications  ... interest_app_dev  \
87                    No                        No  ...               No   
88                    No                        No  ...              Yes   
89                    No                        No  ...               No   
90                    No                        No  ...               No   
91                    No                        No  ...               No   

   interest_data_analytics interest_machine_learning interest_programming  \
87                      No                        No                  Yes   
88                      No                        No                   No   
89                      No                       Yes                   No   
90                      No                        No                   No   
91                      No                       Yes                   No   

   interest_game_design interest_hardware_repair gender  \
87                  Yes                       No  Woman   
88                   No                       No  Woman   
89                  Yes                       No  Woman   
90                   No                       No    Man   
91                   No                      Yes  Woman   

                        race_ethnicity    age               major_mapped  
87  Hispanic or Latino;White/Caucasian  21-24  Early Childhood Education  
88                  Hispanic or Latino  19-20               Liberal Arts  
89              Black/African American  19-20               Liberal Arts  
90                  Hispanic or Latino  19-20                    Nursing  
91                               Asian  21-24   Child and Family Studies  

[5 rows x 44 columns]

<h5 font-weight:bold; font-family:Arial;">Citations</h5>
<p>https://chatgpt.com/share/68e9ce82-b2d4-800b-a0ee-e402ed60433b</p>
<p>https://support.microsoft.com/en-us/office/top-ten-ways-to-clean-your-data-2844b620-677c-47a7-ac3e-c2e157d1db19</p>

In [ ]:
df.to_csv('data/cleaned_data.csv', index=False)